supervised/preprocessing/eda.py

```diff
@@ -162,3 +160,161 @@ def compute(X, y, eda_path):
+                for col in X.columns:
+
+                    if PreprocessingUtils.get_type(X[col]) == "continous":
+
+                        plt.figure(figsize=(5, 5))
+                        plt.scatter(X[col].values, y)
+                        plt.gca().set_xlabel(f"{col}")
+                        plt.gca().set_ylabel("target")
+                        plt.gca().set_title(
+                            f"Scatter plot of {col} vs target",
+                            fontsize=11,
+                            weight="bold",
+                            alpha=0.75,
+                        )
+
+                        plt.savefig(os.path.join(save_path, f"{col}_target"))
+
+                    elif PreprocessingUtils.get_type(X[col]) in (
+                        "categorical",
+                        "discrete",
+                    ):
+                        if X[col].nunique() <= 7:
+
+                            plt.figure(figsize=(5, 5))
+                            for i in X[col].unique():
+                                sns.kdeplot(
+                                    y[X[X[col] == i].index],
+                                    shade=True,
+                                    label=f"{col}_{i}",
+                                )
+                            plt.gca().set_title(
+                                f"Distribution of target for each {col}",
+                                fontsize=11,
+                                weight="bold",
+                                alpha=0.75,
+                            )
+                            plt.legend()
+
+                            plt.savefig(os.path.join(save_path, f"{col}_target"))
+
+                    elif PreprocessingUtils.get_type(X[col]) == "datetime":
+
+                        plt.figure(figsize=(5, 5))
+                        plt.plot(X[col], y)
+                        plt.gca().set_xticklabels(X[col].dt.date, rotation="45")
+                        plt.gca().set_title(
+                            f"Distribution of target over time",
+                            fontsize=11,
+                            weight="bold",
+                            alpha=0.75,
+                        )
+                        plt.savefig(os.path.join(save_path, f"{col}_target"))
+
+            cols = [
+                col
+                for col in X.columns
+                if PreprocessingUtils.get_type(X[col]) == "continous"
+            ][:COLS]
+            if len(cols) > 0:
+                X["target"] = y
+                plt.figure(figsize=(10, 10))
+                sns.heatmap(X[cols + ["target"]].corr())
+                plt.gca().set_title(
+                    "Heatmap", fontsize=11, weight="bold", alpha=0.75,
+                )
+
+            plt.savefig(os.path.join(save_path, "heatmap"))
+
+            with open(os.path.join(save_path, "Extensive_EDA.md"), "w") as fout:
+
+                for col in X.columns[:-1]:
```

Comment:
Why do you exclude the last column?

Reply:
@pplonski The last column is the `target` feature which is added to calculate to correlation to plot heatmap. Hence it's ignored while writing output reports.